src/main/webapp/3Dscript.js

```diff
@@ -45,144 +49,285 @@ function initThreeJs() {
+ * point for the 2D map.
+ * @return {array} An array containing median lat, lng, and alt
+ *   coordinates.
+ */
+function findMedians() {
+  const arrays = {lat: [], lng: [], alt: []};
+  for (const point of pose) {
+    arrays.lat.push(point.lat);
+    arrays.lng.push(point.lng);
+    arrays.alt.push(point.alt);
+  }
+  const sorted = {
+    lat: [...arrays.lat].sort((a, b) => a - b),
+    lng: [...arrays.lng].sort((a, b) => a - b),
+    alt: [...arrays.alt].sort((a, b) => a - b)};
+  const mid = {
+    lat: sortedLat[Math.floor(sorted.lat.length / 2)],
+    lng: sortedLng[Math.floor(sorted.lng.length / 2)],
+    alt: sortedAlt[Math.floor(sorted.alt.length / 2)]};
+  return [mid.lat, mid.lng, mid.alt];
 }
 
 /**
  * Creates the 2D terrain and points a light at it.
- * It also calls two other functions so it can wait on the fetch 
- * request and significantly cut down rendering time with 
+ * It also calls two other functions so it can wait on the fetch
+ * request and significantly cut down rendering time with
  * geometry instancing.
 */
 function addMap() {
+  const [medianLat, medianLng, medianAlt] = findMedians();
+  median = {lat: medianLat, lng: medianLng, alt: medianAlt};
+
   // Creates the 2D map.
-  let loader = new THREE.TextureLoader();
-  let material = new THREE.MeshLambertMaterial({ 
+  const loader = new THREE.TextureLoader();
+  const material = new THREE.MeshLambertMaterial({
     map: loader.load(
-      'https://maps.googleapis.com/maps/api/staticmap' +
-      '?format=png&center=' + pose[0].lat + ',' + pose[0].lng + 
-      '&zoom=18&size=500x500&key=' + apiKey)
-    });
-  let geometry = new THREE.PlaneGeometry(50,50);
+        'https://maps.googleapis.com/maps/api/staticmap' +
+      '?format=png&center=' + median.lat + ',' + median.lng +
+      '&zoom=18&size=500x500&key=' + apiKey),
+  });
+  const geometry = new THREE.PlaneGeometry(200, 200);
   const map = new THREE.Mesh(geometry, material);
-  map.position.set(0,-4,0);
-  map.rotation.x = -Math.PI / 2;
+  map.position.set(0, 0, -4);
   scene.add(map);
 
   // Add the light to the scene.
-  let light = new THREE.PointLight(0xffffff, 1, 0 );
-  light.position.set(0, 100, 0);
-  scene.add(light)
+  const light = new THREE.PointLight(0xffffff, 1, 0 );
+  light.position.set(0, 0, 1000);
+  scene.add(light);
+
+  createInstances();
   plotTrajectory();
-  
-  // Adds pose data to the scene as one Geometry instance.
-  material = new THREE.MeshBasicMaterial({color: 'red'});
-  geometry = new THREE.CylinderBufferGeometry(.005, .005, .1);
-  orientation = new THREE.InstancedMesh(geometry, material, pose.length);
-  orientation.instanceMatrix.setUsage(THREE.DynamicDrawUsage);
-  scene.add(orientation);
   plotOrientation();
 }
 
+
 /**
- * This uses the pose data to create a blue line representing the pose
- * trajectory. 
+ * Creates all the children for each axis of orientation. Each object
+ * is a geometry instance, initializing the object with one geometry
```

Comment:
is each object referring to the entire mesh or like each individual cylinder/point? wondering bc of click on pose implementation considerations

Reply:
Each object refers to the 3 instanced meshes which each have thousands of colored cylinders. Each point now looks something like this -> https://i.stack.imgur.com/6ySFH.png